Diagnostics/HealthChecker/Analyzer/Test-IanaTimeZoneMapping.ps1

```diff
@@ -0,0 +1,79 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. $PSScriptRoot\..\..\..\Shared\ErrorMonitorFunctions.ps1
+
+function Test-IanaTimeZoneMapping {
+    [CmdletBinding(DefaultParameterSetName = "FilePath")]
+    param(
+        [Parameter(Mandatory = $true, ParameterSetName = "FilePath")]
+        [string]$FilePath,
+
+        [Parameter(Mandatory = $true, ParameterSetName = "MappingFile")]
+        [System.Xml.XmlNode]$IanaMappingFile
+    )
+
+    begin {
+        $xmlMap = New-Object System.Collections.Generic.HashSet[string]
+        $xmlMissingAttributesList = New-Object System.Collections.Generic.HashSet[object]
+        $xmlDuplicateEntriesList = New-Object System.Collections.Generic.List[object]
+    } process {
+
+        if ($PSCmdlet.ParameterSetName -eq "FilePath") {
+            if ((Test-Path -Path $FilePath) -eq $false) {
+                Write-Verbose "Path: $FilePath doesn't exist"
+
+                return
+            }
+
+            try {
+                [xml]$IanaMappingFile = Get-Content -Path $FilePath -ErrorAction Stop
+            } catch {
+                Write-Verbose "Exception while trying to import file: $FilePath - Exception: $_"
+                Invoke-CatchActions
+            }
+        }
```

Comment:
Not that it matters here, but this isn't going to work with remote context.

Reply:
I wanted to not over engineer the solution and kept it as simple as possible. In the other script this can be handled to make it work on remote systems via script block or so. Will start working on this tomorrow or on Thursday